15.database_sql_part_2/aggregation_inner_join_subquery.sql

```diff
@@ -0,0 +1,29 @@
+SELECT 
+    authors.first_name || ' ' || authors.last_name AS author_name,
+    SUM(sales.quantity) AS total_books_sold
+FROM 
+    authors
+INNER JOIN 
+    books ON authors.id = books.author_id
+INNER JOIN 
+    sales ON books.id = sales.book_id
+GROUP BY 
+    authors.id
+HAVING 
+    SUM(sales.quantity) = (
+        SELECT 
+            MAX(total_sales)
+        FROM (
+            SELECT 
+                authors.id AS author_id,
+                SUM(sales.quantity) AS total_sales
+            FROM 
+                authors
+            INNER JOIN 
+                books ON authors.id = books.author_id
+            INNER JOIN 
+                sales ON books.id = sales.book_id
+            GROUP BY 
+                authors.id
+        ) AS subquery
+    );
```

Comment:
we also can do something like this
select a.first_name || ' ' || a.last_name AS author_name, m_total.total_sales from authors as a
join
(SELECT 
    authors.id AS author_id,
    SUM(sales.quantity) AS total_sales
FROM 
    authors
INNER JOIN 
    books ON authors.id = books.author_id
INNER JOIN 
    sales ON books.id = sales.book_id
GROUP BY 
    authors.id
order by total_sales desc 
limit 1) as m_total on m_total.author_id = a.id ;